flux-plugin/src/flux-image-automation/image-automation.tsx

```diff
@@ -0,0 +1,210 @@
+import { K8s } from '@kinvolk/headlamp-plugin/lib';
+import Event from '@kinvolk/headlamp-plugin/lib/k8s/event';
+import { apiFactory } from '@kinvolk/headlamp-plugin/lib/ApiProxy';
+import React from 'react';
+import { useLocation } from 'react-router';
+import {
+  ConditionsTable,
+  DateLabel,
+  MainInfoSection,
+  SectionBox,
+  ShowHideLabel,
+  Table,
+} from '@kinvolk/headlamp-plugin/lib/components/common';
+import { SuspendAction, ResumeAction, SyncAction } from '../actions/index';
+import YAML from 'yaml';
+import Editor from '@monaco-editor/react';
+
+const IMAGE_REPOSITORIES = 'imagerepositories.image.toolkit.fluxcd.io';
+const IMAGE_POLICIES = 'imagepolicies.image.toolkit.fluxcd.io';
+const IMAGE_UPDATE_AUTOMATIONS = 'imageupdateautomations.image.toolkit.fluxcd.io';
+
+export function FluxImageAutomationDetailView() {
+  const location = useLocation();
+  const segments = location.pathname.split('/');
+  // The fourth last segment is the kind
+  const namespace = segments[segments.length - 3];
+  // The second last segment is the type
+  const type = segments[segments.length - 2];
+  // The last segment is the name
+  const name = segments[segments.length - 1];
+
+  function getType() {
+    if (type === 'imagerepositories') {
+      return IMAGE_REPOSITORIES;
+    }
+    if (type === 'imagepolicies') {
+      return IMAGE_POLICIES;
+    }
+    return IMAGE_UPDATE_AUTOMATIONS;
+  }
+
+  function getKind() {
+    if (type === 'imagerepositories') {
+      return 'ImageRepository';
+    }
+    if (type === 'imagepolicies') {
+      return 'ImagePolicy';
+    }
+    return 'ImageUpdateAutomation';
+  }
```

Comment:
The `IMAGE_REPOSITORIES` and similar vars is only used once, AFAICS. And you do a similar logic for the kind.
So why not have an object with them:
```typescript
const fluxImageInfo = {
  imagerepositories: {
    type: 'imagerepositories.image.toolkit.fluxcd.io',
    kind: 'ImageRepository',
  },
  ...
}
```

I think this becomes easier to maintain.